### PR TITLE
feat(plugins/bk-user-restriction): add new plugin

### PR DESCRIPTION
**Pull request by wklken**
### Description

- add bk-user-restriction plugin

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Comment by wklken** [comment]
The pull request [#85](https://github.com/TencentBlueKing/blueking-apigateway-apisix/pull/85) titled "feat(plugins/bk-user-restriction): add new plugin" introduces a new plugin called `bk-user-restriction`. 

### Summary:
- Adds the `bk-user-restriction` plugin.
- Includes related unit tests.
- Ensures code style compliance.
- Verified in a local development environment.

The pull request contains 536 additions across 5 files and has 3 commits.

**Review by Han-Ya-Jun — APPROVED** [review]
LGTM

File: src/apisix/tests/test-bk-user-restriction.lua [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            ctx.var.user.username = "unknown_user"
+            local code = plugin.access(conf, ctx)
+            assert.is_equal(code, 403)
+            assert.is_not_nil(ctx.var.bk_apigw_error)
+            assert.is_true(core.string.find(ctx.var.bk_apigw_error.error.message, 'Request rejected by bk-user restriction') ~= nil)
+            assert.is_true(core.string.find(ctx.var.bk_apigw_error.error.message, 'message="The bk-user is not allowed"') ~= nil)
+            assert.is_true(core.string.find(ctx.var.bk_apigw_error.error.message, 'bk_username="unknown_user"') ~= nil)
+        end)
+
+        it("should allow user not in whitelist", function()
```
Comment (piglei) [inline src/apisix/tests/test-bk-user-restriction.lua:100]: 这里应该是 `should allow user in whitelist `

File: src/apisix/plugins/bk-user-restriction.lua [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            },
+            message = {
+                type = "string",
+                default = "The bk-user is not allowed",
+                minLength = 1,
+                maxLength = 1024,
+            },
+        },
+        oneOf = {
+            { required = { "whitelist" } },
```
Comment (piglei) [inline src/apisix/plugins/bk-user-restriction.lua:44]: 是不是应该在什么地方写插件的文档使用说明，比如我看到这就会好奇两个字段如果都配置会怎么样，理论上字段有互斥。（看了代码的写法，黑名单的优先级是更好，即使 user 命中白名单，仍然可能被黑名单拦下来）

**Pull request merged**